source/agora/common/BlockStorage.d

```diff
@@ -817,6 +822,7 @@ public class MemBlockStorage : IBlockStorage
     {
         this.height_idx = new IndexHeight();
         this.hash_idx = new IndexHash();
+        this.saveBlock(GenesisBlock);
```

Comment:
Hmm.. I need to add the genesis block's UTXOs to the UTXO set. But I'll work on this in my PR after we rebase yours.

Reply:
What I mean is, the Ledger class will manager the UTXO set, not the BlockStorage.

Reply:
Well, since you can assume the BlockStorage always have at least one block, can't you just load it from there ?